ZConfig/cfgparser.py

```diff
@@ -175,7 +170,8 @@ def replace(self, text):
             raise
 
     def error(self, message):
-        raise ZConfig.ConfigurationSyntaxError(message, self.url, self.lineno)
+        v = ZConfig.ConfigurationSyntaxError(message, self.url, self.lineno)
+        reraise(type(v), v, sys.exc_info()[2])
```

Comment:
Yes, this is to keep the original traceback. There are *lots* of places in the code that do `try:...except: self.error()`. Debugging those was a nightmare without keeping the original traceback. 

The simplest textual change was to change the (relatively few) `error` functions instead of changing all the callers.